File: java/src/main/java/com/microsoft/sample/aad/protectedapi/security/AuthContext.java
package com.microsoft.sample.aad.protectedapi.security;

import java.util.List;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
public class AuthContext {
  private String audience;
  private String requesterId;
  private String requiredRole;
  private Boolean isThirdPartyAccessAttempt;
  private List<String> roles;
  private String requestedClientId;
  private List<String> clientRoles;
  private String apiClientId;
  private String tenantId;
}

File: dotnet/Security/AuthContext.cs
using System.Collections.Generic;

namespace dotnet.Security
{
    public class AuthContext
    {
        public string Audience { get; set; }
        public string RequesterId { get; set; }
        public string RequiredRole { get; set; }
        public bool IsThirdPartyAccessAttempt { get; set; }
        public List<string> Roles { get; set; }
        public string RequestedClientId { get; set; }
        public List<string> ClientRoles { get; set; }
        public string ApiClientId { get; set; }
        public string TenantId { get; set; }
        public bool Authorized { get; set; }
        public string ForbidReason { get; set; }
    }
}